### PR TITLE
Deprecate `ButtonBar`, `ButtonBarThemeData`, and `ThemeData.buttonBarTheme`

### LINKED ISSUE #127955: Deprecate `ButtonBar`
As can be seen in https://github.com/flutter/flutter/issues/79171, the `ButtonBar` doesn't support the new button types, and it was promised that it would soon be deprecated. That was in April 2021.

It still hasn't been deprecated, and it still doesn't support the new button types.

It is slightly confusing to work with a class that doesn't support an aspect of Flutter, with no indication that it doesn't do so.

I'm not sure if it's gonna stay or not, but I thought I'd highlight the issue.

### PR DESCRIPTION
**Pull request by TahaTesser**
fixes [Deprecate `ButtonBar`](https://github.com/flutter/flutter/issues/127955)

### Code sample

<details>
<summary>expand to view the code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(
        buttonBarTheme: const ButtonBarThemeData(
          alignment: MainAxisAlignment.spaceEvenly,
        ),
      ),
      home: Scaffold(
        body: ButtonBar(
          alignment: MainAxisAlignment.spaceEvenly,
          children: <Widget>[
            TextButton(
              onPressed: () {},
              child: const Text('Button 1'),
            ),
            TextButton(
              onPressed: () {},
              child: const Text('Button 2'),
            ),
            TextButton(
              onPressed: () {},
              child: const Text('Button 3'),
            ),
          ],
        ),
      ),
    );
  }
}
```

</details>

### Before executing `dart fix --apply`
```dart
  return MaterialApp(
      theme: ThemeData(
        buttonBarTheme: const ButtonBarThemeData(
          alignment: MainAxisAlignment.spaceEvenly,
        ),
      ),
      home: Scaffold(
        body: ButtonBar(
          alignment: MainAxisAlignment.spaceEvenly,
          children: <Widget>[
            TextButton(
              onPressed: () {},
              child: const Text('Button 1'),
            ),
            TextButton(
              onPressed: () {},
              child: const Text('Button 2'),
            ),
            TextButton(
              onPressed: () {},
              child: const Text('Button 3'),
            ),
          ],
        ),
      ),
    );
```

### After executing `dart fix --apply`
```dart
    return MaterialApp(
      theme: ThemeData(

      ),
      home: Scaffold(
        body: OverflowBar(
          alignment: MainAxisAlignment.spaceEvenly,
          children: <Widget>[
            TextButton(
              onPressed: () {},
              child: const Text('Button 1'),
            ),
            TextButton(
              onPressed: () {},
              child: const Text('Button 2'),
            ),
            TextButton(
              onPressed: () {},
              child: const Text('Button 3'),
            ),
          ],
        ),
      ),
    );
```

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

File: packages/flutter/lib/fix_data/fix_material/fix_button_bar.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+#     * AppBar: fix_app_bar.yaml
+#     * ButtonBar: fix_button_bar.yaml
+#     * ColorScheme: fix_color_scheme.yaml
+#     * Material (general): fix_material.yaml
+#     * SliverAppBar: fix_sliver_app_bar.yaml
+#     * TextTheme: fix_text_theme.yaml
+#     * ThemeData: fix_theme_data.yaml
+version: 1
+transforms:
+  # Changes made in TBD.
```
Comment (Piinks) [inline packages/flutter/lib/fix_data/fix_material/fix_button_bar.yaml:30]: Here and elsewhere, update TBDs.

File: packages/flutter/lib/src/material/theme_data.dart [hunk 2]
```diff
@@ -538,6 +541,8 @@ class ThemeData with Diagnosticable {
     timePickerTheme ??= const TimePickerThemeData();
     toggleButtonsTheme ??= const ToggleButtonsThemeData();
     tooltipTheme ??= const TooltipThemeData();
+    // DEPRECATED (newest deprecations at the bottom)
+    buttonBarTheme ??= const ButtonBarThemeData();
```
Comment (Piinks) [inline packages/flutter/lib/src/material/theme_data.dart:545]: Ah! Just when ThemeData was clear! 😆

**Review by Piinks — COMMENTED** [review]
Just throwing this idea out there since OverflowBar does not have a theme, do you think there is any value in refactoring ButtonBar to use the OverflowBar internally?

**Review by farisbasha — COMMENTED** [review]
seems oke

File: packages/flutter/lib/src/material/theme_data.dart (hunk 2 above)
Comment (TahaTesser) [reply packages/flutter/lib/src/material/theme_data.dart:545]: Haha yes, i saw your comment [here](https://github.com/flutter/flutter/pull/144078/files#r1515110959) when I had to restore all the deprecation sections with comments.

**Comment by TahaTesser** [comment]
@Piinks 
If you agree, I can update this PR to use `OverflowBar` as the underlying widget for `ButtonBar` and update migration guide  on Flutter website repo to only deprecate unused `ButtonBarTheme` properties.

**Comment by Piinks** [comment]
Thank Taha for considering the keep-it path and the additional context. We discussed a bit more offline and it makes more sense to deprecate this. 👍  Just wanted to be extra considerate before we did so.

**Review by Piinks — APPROVED** [review]
Just the TBDs in the fix data files are left to fill in here, otherwise, this LGTM! We can land the migration guide once this is in.

**Pull request merged**